scraper/load.py

```diff
@@ -0,0 +1,526 @@
+        features = product_dict["features"]
+
+        for feature, value in features.items():
+            if feature in self.features_map:
+                self.add_feature_value(value, feature)
+                continue
+
+            feature_xml = f"""<prestashop xmlns:xlink="http://www.w3.org/1999/xlink">
+                    <product_feature>
+                        <name>
+                            <language id="1"><![CDATA[{feature}]]></language>
+                        </name>
+                    </product_feature>
+                </prestashop>
+            """
+
+            response = self.make_request("POST", "product_features", feature_xml)
+
+            if response.status_code == 201:
+                response_xml = ET.fromstring(response.text)
+                feature_id = response_xml.find('.//id').text
+
+                print("Feature dodany.")
+                print(f"ID feature: {feature_id}")
+                self.features_map[feature] = {}
+                self.features_map[feature]["id"] = feature_id
+                self.features_map[feature]["values"] = {}
+                self.add_feature_value(value, feature)
+            else:
+                print("Failed to upload feature:", response.status_code, response.text)
+
+    def add_feature_value(self, value, feature):
+        if value in self.features_map[feature]["values"]:
+            return
+
+        feature_xml = f"""<?xml version="1.0" encoding="UTF-8"?>
+            <prestashop xmlns:xlink="http://www.w3.org/1999/xlink">
+                <product_feature_value>
+                    <id_feature><![CDATA[{self.features_map[feature]["id"]}]]></id_feature>
+                    <value>
+                        <language id="1"><![CDATA[{value}]]></language>
+                    </value>
+                </product_feature_value>
+            </prestashop>
+            """
+
+        response = self.make_request("POST", "product_feature_values", feature_xml)
+
+        if response.status_code == 201:
+            response_xml = ET.fromstring(response.text)
+            feature_value_id = response_xml.find('.//id').text
+
+            print("Feature value dodany.")
+            print(f"ID feature value: {feature_value_id}")
+            self.features_map[feature]["values"][value] = feature_value_id 
+
+        else:
+            print("Failed to upload feature value:", response.status_code, response.text)
+
+    def make_request(self, method, route, xml):
+        response = None
+        if method == "POST":
+            response = requests.post(
+                    self.api_url + route,
+                    data=xml.encode('utf-8'),
+                    headers={'Content-Type': 'application/xml'},
+                    auth=(self.api_key, '')
+                )
+        return response
+
+
+if __name__ == "__main__":
+    api_url = "http://localhost:8000/api/"
+    api_key = os.getenv("api_key")
+    RESULTS_PAGES = 1 #21
```

Comment:
It would be nice if this script could automatically read all results available in `results` directory